subsys/bluetooth/host/mesh/prov.c

```diff
@@ -1067,20 +1067,20 @@ static void prov_data(const u8_t *data)
 	link.expect = 0;
 
 	/* Store info, since bt_mesh_provision() will end up clearing it */
-	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY)) {
+#ifdef CONFIG_BT_MESH_GATT_PROXY
 		identity_enable = is_pb_gatt();
-	} else {
+#else
 		identity_enable = false;
-	}
+#endif
 
 	bt_mesh_provision(pdu, net_idx, flags, iv_index, addr, dev_key);
 
 	/* After PB-GATT provisioning we should start advertising
 	 * using Node Identity.
 	 */
-	if (identity_enable) {
+#ifdef CONFIG_BT_MESH_GATT_PROXY
```

Comment:
Could you instead try changing this to `if (IS_ENABLED(CONFIG_BT_GATT_PROXY) && identity_enable)` and see if that also fixes the issue? (it'd be a less intrusive change)

Reply:
This actually fixes the issue. i updated the PR with your suggestions.